main.js

```diff
@@ -674,3 +684,91 @@ window.onload = () => {
     document.getElementById("ambient-encounter-link").addEventListener("click", showAmbientEncounter);
     document.getElementById("dictated-note-link").addEventListener("click", showDictatedNote);
 }
+
+// Note customization ----------------------------------------------------------------
+
+const templateSectionsMap = {
+    "GENERIC_MULTIPLE_SECTIONS": [
+        "CHIEF_COMPLAINT",
+        "HISTORY_OF_PRESENT_ILLNESS",
+        "PAST_MEDICAL_HISTORY",
+        "PAST_SURGICAL_HISTORY",
+        "PAST_OBSTETRIC_HISTORY",
+        "FAMILY_HISTORY",
+        "SOCIAL_HISTORY",
+        "ALLERGIES",
+        "CURRENT_MEDICATIONS",
+        "IMMUNIZATIONS",
+        "VITALS",
+        "LAB_RESULTS",
+        "IMAGING_RESULTS",
+        "PHYSICAL_EXAM",
+        "ASSESSMENT",
+        "PLAN",
+        "PRESCRIPTION",
+        "APPOINTMENTS"
+    ],
+    "GENERIC_SOAP": [
+        "SUBJECTIVE",
+        "OBJECTIVE",
+        "ASSESSMENT",
+        "PLAN"
+    ],
+};
+
+const initPage = () => {
+    updateSectionsList();
+}
+
+const onTemplateChange = () => {
+    noteSectionsCustomization = {};
+    updateSectionsList();
+}
+
+const updateSectionsList = () => {
+    const template = getNoteTemplate();
+    const selectElement = document.getElementById("note-sections");
+    selectElement.innerHTML = "";
+
+    const sections = templateSectionsMap[template] || [];
+    sections.forEach((sectionKey) => {
+        const opt = document.createElement("option");
+        opt.value = sectionKey;
+        opt.innerText = sectionKey;
+        selectElement.appendChild(opt);
+    });
+
+    selectElement.value = sections[0] || "";
+    onSectionChange();
+}
+
+const onSectionChange = () => {
+    const selected = document.getElementById("note-sections").value;
+    if (!selected) {
+        document.getElementById("section-customization-fields").style.display = "none";
+        return;
+    }
+
+    document.getElementById("section-customization-fields").style.display = "inline-block";
+    const existing = noteSectionsCustomization[selected] || {};
+    document.getElementById("style-select").value = existing.style || "";
+    document.getElementById("custom-instruction").value = existing.custom_instruction || "";
+}
+
+const onCustomizationChange = () => {
```

Comment:
IMO It's a bit confusing to mix "custom instruction" and "style" configuration in the same function, I would split in 2 separate ones